### PR TITLE
Tactical fix for specialization issue #2393

### PR DESCRIPTION
**Pull request by mathewc**
See https://github.com/Azure/azure-functions-host/issues/2392

### CODE REVIEW

File: src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs [hunk 1]
```diff
@@ -168,11 +168,16 @@ internal void EnsureInitialized(WebHostSettings settings)
...
                         StandbyManager.Initialize(_standbyScriptHostConfig);
 
-                        // start a background timer to identify when specialization happens
-                        // specialization usually happens via an http request (e.g. scale controller
-                        // ping) but this timer is started as well to handle cases where we
-                        // might not receive a request
-                        _specializationTimer = new Timer(OnSpecializationTimerTick, settings, 1000, 1000);
+                        // TEMP - when in Zip deploy mode, temporarily disabling timer based specialization
+                        // until issue https://github.com/Azure/azure-functions-host/issues/2392 is fixed.
+                        if (!_settingsManager.IsZipDeployment)
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs]: Actually there is a problem with this - I'll update the PR in a minute

File: src/WebJobs.Script/Config/ScriptSettingsManager.cs [hunk 2]
```diff
@@ -110,6 +112,11 @@ public virtual void Reset()
             _settingsCache.Clear();
         }
 
+        public virtual bool GetBooleanSetting(string settingKey)
+        {
+            return GetSetting(settingKey) == "1";
```
Comment (davidebbo) [inline src/WebJobs.Script/Config/ScriptSettingsManager.cs]: Won't this only work if it's set to "1"? We need to check whether this App Setting is set at all.

File: src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs [hunk 3]
```diff
@@ -267,6 +267,13 @@ private TInstance GetActiveInstance<TInstance>(WebHostSettings settings, ref TIn
 
         private void OnSpecializationTimerTick(object state)
         {
+            if (_settingsManager.IsZipDeployment)
+            {
+                // TEMP - when in Zip deploy mode, temporarily disabling timer based specialization
+                // until issue https://github.com/Azure/azure-functions-host/issues/2392 is fixed.
+                return;
+            }
```
Comment (davidebbo) [inline src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs]: Before your previous change, I think we already had the first EnsureInitialized, but not the second. Just want to make sure that we're ok with not having the first at all.

Comment (mathewc) [reply src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs]: We want to skip all of this. This OnSpecializationTimer change came in 2 commits, the second being a bug fix adding the second call.

For this tactical fix, we're saying that when in Zip mode we're relying on the http stimulus to trigger specialization.

**Pull request merged**